File: web/classes/exception.php
<?php

final class NM_exception extends Exception
{
    public function __construct($code, $msg, $file, $line)
    {
        $this->code = $code;
        $this->message = $msg;
        $this->file = $file;
        $this->line = $line;
    }
}

File: classes/exception.cpp
#include "exception.hpp"

using namespace std;

NM_exception::NM_exception(NM_exception_code code, string message)
{
    this->init(code, message);
}

void NM_exception::init(NM_exception_code code, string message)
{
    this->code = code;
    this->message = message;
}

NM_exception_code NM_exception::get_code()
{
    return this->code;
}

string NM_exception::get_message()
{
    return this->message;
}
